runtime/src/main/java/com/flipkart/flux/deploymentunit/DeploymentUnitUtil.java

```diff
@@ -0,0 +1,212 @@
+        Set<Method> methods = new HashSet<>();
+
+        //loading this class separately in this class loader as the following isAnnotationPresent check returns false, if
+        //we use default class loader's Workflow, as both class loaders don't have any relation between them.
+        Class workflowClass = urlClassLoader.loadClass(Workflow.class.getCanonicalName());
+
+        for(String name : classNames) {
+            Class clazz = urlClassLoader.loadClass(name);
+
+            for(Method method : clazz.getMethods()) {
+                if(method.isAnnotationPresent(workflowClass))
+                    methods.add(method);
+            }
+        }
+
+        return methods;
+    }
+
+    /**
+     * Given a class loader, retrieves workflow classes names from config file, and returns methods
+     * which are annotated with {@link com.flipkart.flux.client.model.Task} annotation in those classes.
+     * @param urlClassLoader
+     * @return set of Methods
+     * @throws ClassNotFoundException
+     */
+    public Set<Method> getTaskMethods(URLClassLoader urlClassLoader) throws ClassNotFoundException, IOException {
+
+        YamlConfiguration yamlConfiguration = getProperties(urlClassLoader);
+        List<String> classNames = (List<String>) yamlConfiguration.getProperty(WORKFLOW_CLASSES);
+        Set<Method> methods = new HashSet<>();
+
+        //loading this class separately in this class loader as the following isAnnotationPresent check returns false, if
+        //we use default class loader's Task, as both class loaders don't have any relation between them.
+        Class taskClass = urlClassLoader.loadClass(Task.class.getCanonicalName());
+
+        for(String name : classNames) {
+            Class clazz = urlClassLoader.loadClass(name);
+
+            for(Method method : clazz.getMethods()) {
+                if(method.isAnnotationPresent(taskClass))
+                    methods.add(method);
+            }
+        }
+
+        return methods;
+    }
+
+    /**
+     * Given a deployment unit deployment unit name retrieves properties from config file and returns them.
+     * @param deploymentUnitName
+     * @return
+     * @throws IOException
+     */
+    public YamlConfiguration getProperties(String deploymentUnitName) throws IOException {
+        URLClassLoader classLoader = getClassLoader(deploymentUnitName);
+        return getProperties(classLoader);
+    }
+
+    /**
+     * Given a deployment unit classloader retrieves properties from config file and returns them.
+     * @param urlClassLoader
+     * @return YamlConfiguration
+     * @throws IOException
+     */
+    public YamlConfiguration getProperties(URLClassLoader urlClassLoader) throws IOException {
+        return new YamlConfiguration(urlClassLoader.getResource(CONFIG_FILE));
+    }
+
+    /**
+     * Provides {@link java.net.URLClassLoader for a given deployment unit}
+     */
+    static class ClassLoaderProvider {
```

Comment:
This class will be used by only DeploymentUnitUnit and DeploymentUnitUtil heavily depends on it. Also this class provides class loader to specific structure only. So IMO, it makes more sense to keep it as part of DeploymentUnitUtil itself.